src/language-js/print/decorators.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+const { isNonEmptyArray } = require("../../common/util");
+const {
+  builders: { line, hardline },
+} = require("../../document");
+const { locStart } = require("../loc");
+const {
+  getParentExportDeclaration,
+  hasNewlineBetweenOrAfterDecorators,
+  isExportDeclaration,
+} = require("../utils");
+
+function printDecorators(path, options, print) {
+  const node = path.getValue();
+  const parentExportDecl = getParentExportDeclaration(path);
+  const parts = [];
+  if (
+    isNonEmptyArray(node.decorators) &&
+    // If the parent node is an export declaration and the decorator
+    // was written before the export, the export will be responsible
+    // for printing the decorators.
+    !(
+      parentExportDecl &&
+      locStart(parentExportDecl, { ignoreDecorators: true }) >
+        locStart(node.decorators[0])
+    )
+  ) {
+    const shouldBreak =
+      node.type === "ClassExpression" ||
+      node.type === "ClassDeclaration" ||
+      hasNewlineBetweenOrAfterDecorators(node, options);
+
+    const separator = shouldBreak ? hardline : line;
+
+    path.each((decoratorPath) => {
+      let decorator = decoratorPath.getValue();
+      if (decorator.expression) {
+        decorator = decorator.expression;
+      } else {
+        decorator = decorator.callee;
```

Comment:
This looks like untested code according to Codecov (??)

Reply:
Should be something about

```
@decorator
```

and

```
@decorator()
```

Not sure.

Reply:
This is just unused remnants of some old code.

Reply:
I'm not sure if it can be removed.

Reply:
I'll try to contribute a test for this when I find a spare cycle,

Reply:
I checked `flow` `babel` and `typescript` parser, seems they all use `.expression`, I removed this line.

Reply:
I checked `flow` `babel` and `typescript` parser, seems they all use `.expression`, I removed this line.